### PR TITLE
Fix incorrect glyph page index in FreeTypeFontGenerator

### PR DESCRIPTION
**Pull request by raeleus**
When using the BitmapFontWriter, fonts that require more than one texture page to fit all their glyphs fail. The resulting FNT file points to glyphs on the wrong page, resulting in a font that renders with odd looking characters. This occurs frequently with large fonts and with glyphs that could potentially land on the last row of the first page.

After some investigation, I discovered that it was a minor issue in FreeTypeFontGenerator causing this. It assigns the page index of each glyph to the last index of `packer.getPages()`. This is wrong because glyphs may still fit on the previous page depending on its dimensions.

The fix just applies the glyph hash code as the pixmap name to find the correct page. It's an inelegant solution, but does not require any API changes to adopt.

### CODE REVIEW

**Review by Frosty-J — APPROVED** [review]
Looks good. This will alleviate the amount of untold suffering.

**Comment by raeleus** [comment]
I had no idea it was such a pervasive problem! Thanks, James.

File: extensions/gdx-freetype/src/com/badlogic/gdx/graphics/g2d/freetype/FreeTypeFontGenerator.java [hunk 1]
```diff
@@ -598,8 +598,9 @@ public FreeTypeBitmapFontData generateData (FreeTypeFontParameter parameter, Fre
 			}
 		}
 
-		Rectangle rect = packer.pack(mainPixmap);
-		glyph.page = packer.getPages().size - 1; // Glyph is always packed into the last page for now.
+		String pixmapName = Integer.toString(glyph.hashCode());
```
Comment (tommyettinger) [inline extensions/gdx-freetype/src/com/badlogic/gdx/graphics/g2d/freetype/FreeTypeFontGenerator.java:601]: I think you should probably consider using this instead of a not-necessarily-unique hashCode():

```java
		String pixmapName = Integer.toString(glyph.id);
```

The ID should be unique in any font, barring some weirdness with alternate glyphs. The hashCode() is very likely to be unique, but it is the same as calling System.identityHashCode(), and that's neither as fast as it could be nor as unique as we want it to be.

**Review by tommyettinger — CHANGES_REQUESTED** [review]
tl;dr hashCode() isn't guaranteed to be unique, so some Pixmap names might be duplicates using that. The glyph's id should be unique, if I understand it correctly.

**Comment by raeleus** [comment]
I tested ID and it failed with my fonts. Several glyphs evaluate to 0. I did have the same concern about hash code collisions so I should be able to concatenate ID and hashcode. Hopefully that's enough unless you have any better ideas, @tommyettinger.

**Comment by tommyettinger** [comment]
> I tested ID and it failed with my fonts. Several glyphs evaluate to 0. I did have the same concern about hash code collisions so I should be able to concatenate ID and hashcode. Hopefully that's enough unless you have any better ideas, @tommyettinger.

Sounds good to me! Make sure to separate the two strings with a space or underscore or something, because if the id is 0 that's colliding, then a hash of 12345 with an id of 10 and a hash of 123451 with an id of 0 might still collide (obviously extremely rarely) as 1234510. With an underscore, 12345_10 and 123451_0 wouldn't collide. I don't even know if PixmapPacker has problems with duplicate names; I know an atlas permits them for some reason.

**Review by tommyettinger — APPROVED** [review]

**Comment by raeleus** [comment]
Thanks for the guidance, TEt. This one is ready for approval by another team member.

**Review by tommyettinger — APPROVED** [review]
